lambdas/data-migration2/src/granulesAndFiles.ts

```diff
@@ -0,0 +1,269 @@
+    tableNames.granules,
+    knex
+  );
+
+  const bucket = getBucket(file);
+  const key = getKey(file);
+
+  // Map old record to new schema.
+  const updatedRecord: PostgresFile = {
+    bucket,
+    key,
+    granule_cumulus_id: granuleCumulusId,
+    file_size: file.size,
+    checksum_value: file.checksum,
+    checksum_type: file.checksumType,
+    file_name: file.fileName,
+    source: file.source,
+    path: file.path,
+  };
+  await knex(tableNames.files).insert(updatedRecord);
+};
+
+/**
+ * Migrate granule and files from DynamoDB to RDS
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} dynamoRecord
+ * @param {Knex} knex
+ * @returns {Promise<MigrationSummary>} - Migration summary for files
+ */
+export const migrateGranuleAndFilesViaTransaction = async (
+  dynamoRecord: AWS.DynamoDB.DocumentClient.AttributeMap,
+  knex: Knex
+): Promise<MigrationSummary> => {
+  const fileMigrationSummary = {
+    dynamoRecords: 0,
+    success: 0,
+    failed: 0,
+    skipped: 0,
+  };
+
+  // Validate record before processing using API model schema
+  Manager.recordIsValid(dynamoRecord, schemas.granule);
+  const files = dynamoRecord.files;
+  const granuleId = dynamoRecord.granuleId;
+  const collectionId = dynamoRecord.collectionId;
+
+  await knex.transaction(async () => {
+    await migrateGranuleRecord(dynamoRecord, knex);
+    await Promise.all(files.map(async (file : ApiFile) => {
+      fileMigrationSummary.dynamoRecords += 1;
+      try {
+        await migrateFileRecord(file, granuleId, collectionId, knex);
+        fileMigrationSummary.success += 1;
+      } catch (error) {
+        fileMigrationSummary.failed += 1;
+        logger.error(
+          `Could not create file record in RDS for file ${file}`,
+          error
+        );
+      }
+    }));
+  });
+
+  return fileMigrationSummary;
+};
+
+export const migrateGranulesAndFiles = async (
+  env: NodeJS.ProcessEnv,
+  knex: Knex
+): Promise<GranulesAndFilesMigrationSummary> => {
+  const granulesTable = envUtils.getRequiredEnvVar('GranulesTable', env);
+
+  const searchQueue = new DynamoDbSearchQueue({
+    TableName: granulesTable,
+  });
+
+  const granuleMigrationSummary = {
+    dynamoRecords: 0,
+    success: 0,
+    failed: 0,
+    skipped: 0,
+  };
+
+  const fileMigrationSummary = {
+    dynamoRecords: 0,
+    success: 0,
+    failed: 0,
+    skipped: 0,
+  };
+
+  let record = await searchQueue.peek();
+
+  /* eslint-disable no-await-in-loop */
+  while (record) {
+    granuleMigrationSummary.dynamoRecords += 1;
+    try {
+      const granuleFileMigrationSummary = await migrateGranuleAndFilesViaTransaction(record, knex);
+      granuleMigrationSummary.success += 1;
+      fileMigrationSummary.dynamoRecords += granuleFileMigrationSummary.dynamoRecords;
+      fileMigrationSummary.success += granuleFileMigrationSummary.success;
+      fileMigrationSummary.failed += granuleFileMigrationSummary.failed;
```

Comment:
Mentioned in my comment https://github.com/nasa/cumulus/pull/2051/files#r561389823 that I don't particularly like this, and I feel like calculating `fileMigrationSummary` here is messy but I am not sure of a cleaner alternative. Open to ideas

Reply:
@npauzenga This is how I ended up refactoring it based on your suggestions [`5a543a5` (#2051)](https://github.com/nasa/cumulus/pull/2051/commits/5a543a53520b1e96e50a477902a4ffb14fc71446). In some respects, it is cleaner but I also think it makes the transaction obsolete because the `try/catch` now handles skipping writing file records if a granule record fails. In this commit I've linked I haven't removed the knex transaction but, if we go this route, I will definitely remove it since I don't think it's necessary. Let me know what you think!